gui/src/main.rs

```diff
@@ -152,21 +163,49 @@ impl App for Gui {
                 }
             });
             ui.horizontal(|ui| {
-                if ui.button("Params").clicked() {}
-                if ui.button("Headers").clicked() {}
-                if ui.button("Body").clicked() {}
+                if ui.button("Params").clicked() {
+                    self.config.request_window_mode = RequestWindowMode::PARAMS;
+                }
+                if ui.button("Headers").clicked() {
+                    self.config.request_window_mode = RequestWindowMode::HEADERS;
+                }
+                if ui.button("Body").clicked() {
+                    self.config.request_window_mode = RequestWindowMode::BODY;
+                }
             });
         });
         match self.config.request_window_mode {
             RequestWindowMode::BODY => {
+                TopBottomPanel::top("request_panel")
+                    .resizable(true)
+                    .min_height(250.0)
+                    .show(ctx, |ui| {
+                        ScrollArea::vertical().show(ui, |ui| {
+                            ui.add(
+                                TextEdit::multiline(&mut self.config.body_str)
```

Comment:
Pretty nice that we get a formatted code block that you can hit enter in and not lose focus. Only thing left would be to figure out a how to implement a code formatting button to beautify the json.

Reply:
I think I remember the docs page mentioning how to install language packs?